src/webapp/pages/form-page/disease-outbreak-event/utils/applyRulesInFormState.ts

```diff
@@ -0,0 +1,89 @@
+import { Rule } from "../../../../../domain/entities/Rule";
+import {
+    FormFieldState,
+    FormSectionState,
+    FormState,
+    getEmptyValueForField,
+} from "../../../../components/form/FormState";
+
+export function applyRulesInFormState(
+    currentFormState: FormState,
+    updatedField: FormFieldState,
+    formRules: Rule[]
+): FormState {
+    const filteredRulesByFieldId = formRules.filter(rule => rule.fieldId === updatedField.id);
+
+    if (filteredRulesByFieldId.length === 0) {
+        return currentFormState;
+    }
+
+    const formStateWithRulesApplied = filteredRulesByFieldId.reduce((formState, rule) => {
+        switch (rule.type) {
+            case "toggleSectionsVisibilityByFieldValue":
+                return {
+                    ...formState,
+                    sections: formState.sections.map(section =>
+                        toggleSectionVisibilityByFieldValue(section, updatedField.value, rule)
+                    ),
+                };
+            default:
```

Comment:
I do not think we need this default case as rule.type is a typescript type and cannot be anything else.